javascript/ql/src/semmle/javascript/security/dataflow/IndirectCommandArgument.qll

```diff
@@ -61,15 +64,22 @@ private DataFlow::SourceNode argumentList(SystemCommandExecution sys, DataFlow::
  * let args = ["-c", cmd];
  * childProcess.spawn(sh, args, cb);
  * ```
+ * or
+ * ```
+ * let cmd = getCommand();
+ * childProcess.spawn("cmd.exe", ["/c"].concat(cmd), cb);
+ * ```
  */
 predicate isIndirectCommandArgument(DataFlow::Node source, SystemCommandExecution sys) {
-  exists(
-    DataFlow::ArrayCreationNode args, DataFlow::Node shell, string dashC
-  |
+  exists(DataFlow::ArrayCreationNode args, DataFlow::Node shell, string dashC |
     shellCmd(shell.asExpr(), dashC) and
     shell = commandArgument(sys, DataFlow::TypeBackTracker::end()) and
-    args = argumentList(sys, DataFlow::TypeBackTracker::end()) and
     args.getAPropertyWrite().getRhs().mayHaveStringValue(dashC) and
-    source = args.getAPropertyWrite().getRhs()
+    args = argumentList(sys, DataFlow::TypeBackTracker::end()) and
```

Comment:
Nit: lets create and use the unary `argumentList(sys)` to reduce this `::end` duplication.